pkg/storage/interfaces.go

```diff
@@ -62,34 +64,13 @@ type MatchValue struct {
 // to that function.
 type TriggerPublisherFunc func(obj runtime.Object) []MatchValue
 
-// Filter is interface that is used to pass filtering mechanism.
-type Filter interface {
-	// Filter is a predicate which takes an API object and returns true
-	// if and only if the object should remain in the set.
-	Filter(obj runtime.Object) bool
```

Comment:
You could rename this method to Matches and then the predicate will implement this interface.